query_functional_test.sh

```diff
@@ -0,0 +1,98 @@
+#!/bin/bash
+
+# todo: handle retries/communications failures? Hopefully unnecessary with local docker setup
+# todo: fix indentation of grpcurl request
+
+set +ex
+
+if [ "$#" -lt 1 ]; then
+    echo "Usage: query_functional_test.sh TAG"
+    exit 1
+fi
+
+TAG=$1
+
+# Make sure the docker env is clean
+docker rm --force $(docker ps --all -q)
+
+# To run this script locally, you may need to run `docker pull $tag` from the model repo,
+# where your credentials will be populated.
+DOCKER_IMAGE_TAG="us.gcr.io/model-159019/gh:$TAG"
+
+# Import data into graphhopper's internal format. It's necessary to move test data from /web before
+# running import, because import paths are relative to base /graphhopper folder, unlike in `mvn test`
+docker run \
+    -v "$TMPDIR:/graphhopper/transit_data/"\
+    --rm \
+     "$DOCKER_IMAGE_TAG" \
+     /bin/bash -c "cp -r ./web/test-data . && java -Xmx2g -Xms1g -XX:+UseG1GC -XX:MetaspaceSize=100M \
+     -classpath web/target/graphhopper-web-1.0-SNAPSHOT.jar -server com.graphhopper.http.GraphHopperApplication import test_gh_config.yaml"
+
+# Run link-mapping step
+docker run \
+    -v "$TMPDIR:/graphhopper/transit_data/"\
+    --rm \
+    "$DOCKER_IMAGE_TAG" \
+    java -Xmx2g -Xms1g -XX:+UseG1GC -XX:MetaspaceSize=100M \
+    -classpath web/target/graphhopper-web-1.0-SNAPSHOT.jar com.graphhopper.http.GraphHopperApplication gtfs_links test_gh_config.yaml
+
+# Run server in background
+docker run --rm --log-driver=none --name functional_test_server -p 50051:50051 -p 8998:8998 \
+    -v "$TMPDIR:/graphhopper/transit_data/" "$DOCKER_IMAGE_TAG" &
+
+echo "Waiting for graphhopper server to start up"
+sleep 30
```

Comment:
still would like something better than sleeping here, but if 30s consistently works 🤷

Reply:
Yeah, agree that it's not ideal, but I decided to kick this can down the road, at least for now. In my next PR I'll see if I can add something a bit more intelligent here, would be nice to get real confirmation that the container was ready without just blindly waiting and hoping